package/libpiksi/libpiksi/include/libpiksi/endpoint.h

```diff
@@ -40,7 +40,9 @@ typedef struct pk_endpoint_s pk_endpoint_t;
  */
 typedef enum {
   PK_ENDPOINT_PUB,
+  PK_ENDPOINT_PUB_SERVER,
   PK_ENDPOINT_SUB,
+  PK_ENDPOINT_SUB_SERVER,
```

Comment:
If there are no usages of sub being a client (or pub being a server) I think we could make this the default  behavior for each and reduce the number of socket types. Alternately you could add a config enum/param to the endpoint creation function that would change the bind vs connect behavior. Yet another option that would be much more of a "drop in" change would be to duplicate the @ and > functionality (it could be simplified by making it a required part of the connection string).

Reply:
We kind of talked about this 1-1, but basically all router socket use bind whether pub or sub, and other daemons use connect to communicate via the router. I really wanted to get rid of the non-standard '>' and '@' usage so I went with hardcoding our current usage. A larger tech debt refactor of the router might inform a better decision overall on this.